### DISCUSSION TITLE
Filter a multiple relationship field

### DISCUSSION THREAD

**Discussion by moritz4** [pocketbase/pocketbase#5893] [Q&A] [answered]
For the below schema, how can query the posts collection and filter `posts` which have at least one `tag` which has both a certain `name` and `region`.

E.g. (pseudocode) I want posts which have a tag with name = Red and region = Asia.

```
pb.collection("posts").getFullList({
    filter: "tag contains (name = 'Red' AND region = 'Asia')"
});
```

<img width="416" alt="Screenshot 2024-11-18 at 18 39 22" src="https://github.com/user-attachments/assets/4a8be3df-9a5b-4670-b486-9efd6853bc74">

**Comment by ganigeorgiev** [comment, marked as answer]
You can try:

```js
pb.collection("posts").getFullList({
    filter: "tags.name ?= 'Red' && tags.region ?= 'Asia'"
});
```

The above should work because the filter, similar to a `WHERE` clause, acts as a "funnel" where with each following constraint you are narrowing further the results.

Or if a pseudo-SQL would be more easy to understand (_you can print the generated query with `--dev`_):
```sql
SELECT DISTINCT posts.*
FROM posts
LEFT JOIN tags on posts.tags HAS tags.id -- fake HAS operator for simplicity
WHERE tags.name = 'Red' AND tags.region = 'Asia'
```
